src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509KeyUsageExtension.cs

```diff
@@ -40,6 +43,69 @@ public override void CopyFrom(AsnEncodedData asnEncodedData)
             _decoded = false;
         }
 
+        private static byte[] EncodeX509KeyUsageExtension(X509KeyUsageFlags keyUsages)
+        {
+            // The numeric values of X509KeyUsageFlags mean that if we interpret it as a little-endian
+            // ushort it will line up with the flags in the spec. We flip bit order of each byte to get
+            // the KeyUsageFlagsAsn order expected by AsnWriter.
+            KeyUsageFlagsAsn keyUsagesAsn =
+                (KeyUsageFlagsAsn)ReverseBitOrder((byte)keyUsages) |
+                (KeyUsageFlagsAsn)(ReverseBitOrder((byte)(((ushort)keyUsages >> 8))) << 8);
+
+            // The expected output of this method isn't the SEQUENCE value, but just the payload bytes.
+            // We expect to encode at most 9 bits in the bit list, which can encode to at most 5 octets: two
+            // for the tag and definite length, two for the bit string, and one for the unused bit count.
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER, initialCapacity: 5);
```

Comment:
During code shuffling, I added this `initialCapacity` so the internal allocation is smaller. Aside from this (and member modifiers) nothing else changed.